source/views/help/wifi.js

```diff
@@ -54,35 +55,57 @@ function reportToServer(data) {
 type Props = {}
 
 type State = {
-  status: string,
+  error: ?string,
+  status: 'init' | 'collecting' | 'reporting' | 'done' | 'error',
 }
 
 export class ReportWifiProblemView extends React.Component<Props, State> {
   state = {
-    status: '',
+    error: null,
+    status: 'init',
   }
 
   start = async () => {
-    this.setState(() => ({status: 'Collecting data…'}))
+    this.setState(() => ({status: 'collecting', error: ''}))
     const [position, device] = await Promise.all([getPosition(), collectData()])
-    this.setState(() => ({status: 'Reporting data…'}))
+    this.setState(() => ({status: 'reporting'}))
     try {
       let data = {position, device, version: 1}
       await retry(() => reportToServer(data), {retries: 10})
       await delay(1000)
-      this.setState(() => ({status: 'Thanks!'}))
+      this.setState(() => ({status: 'done'}))
     } catch (err) {
       reportNetworkProblem(err)
       this.setState(() => ({
-        status: 'Apologies; there was an error. Please try again later.',
+        error: 'Apologies; there was an error. Please try again later.',
+        status: 'error',
       }))
     }
   }
 
   render() {
+    const buttonEnabled =
+      this.state.status === 'init' || this.state.status === 'error'
+
+    let buttonMessage = 'Error'
+    if (this.state.status === 'init') {
```

Comment:
Can this `if`-`else if` chain be replaced by a switch-case?

Reply:
It can, but then I have to add a `break` for each case.

Reply:
Hmm, what about a function above that converts `status` → `message`?

Reply:
Case statement would be nice, but I can live with either.

Reply:
A function like I suggested could use switch-case and would also reduce the size of this render function. (Though, it would be a new call.)  Alternatively, and perhaps even better, this could be a map?  E.g.,
```js
const buttonMessages = {
  'init': 'Report',
  // ...
}

// ...

let buttonMessage = buttonMessages[this.state.status] || 'Error'
```

IDK if that's good coding style in JS, but that would handle the same cases and allow for a denser representation of the possible mappings.

Reply:
I'm okay with doing an object. Flow even ensures that we don't use a property that doesn't exist on it.